drivers/net/tun.c

```diff
@@ -989,7 +988,7 @@ static ssize_t tun_write(FAR struct file *filep, FAR const char *buffer,
       ret = nxmutex_lock(&priv->lock);
       if (ret < 0)
         {
-          return nwritten == 0 ? (ssize_t)ret : nwritten;
```

Comment:
the change is wrong, you need consider the partial write.

Reply:
@xiaoxiang781216 In short, all cases with `nwritten != 0` will break the `for` loop, and only `nwritten == 0` can reach here.

At first, we were unaware of this, and the Coverity scan told us it's a dead code.

Then we found the original [change](https://github.com/apache/nuttx/pull/670/files#diff-0b448059b6f4c39cbe8bfa499ac018d474f2dee54ff35ef79ec6c4e9f030b0c9R1218) just added a useless `nwritten`, which will only be 0 in the loop and always break/return when it's not 0. So does the `nread`.
@zs39 Maybe you need to add more description in the commit message and in the pull request about this.

BTW, If we have a problem with partial write, it should be a historical problem and not introduced by this change, but shall TUN support partial write?

Reply:
Ok, if the old code doesn't support the partial write/read, let's add it later if the need is raised.